virtual-hubs-network-assigner/src/main/java/com/farao_community/farao/virtual_hubs/network_extension_builder/VirtualHubAssigner.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+package com.farao_community.farao.virtual_hubs.network_extension_builder;
+
+import com.farao_community.farao.virtual_hubs.VirtualHub;
+import com.farao_community.farao.virtual_hubs.network_extension.AssignedVirtualHubAdder;
+import com.powsybl.iidm.network.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * @author Baptiste Seguinot {@literal <baptiste.seguinot@rte-france.com>}
+ */
+public class VirtualHubAssigner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VirtualHubAssigner.class);
+    private List<VirtualHub> virtualHubs;
+
+    public VirtualHubAssigner(List<VirtualHub> virtualHubs) {
+        this.virtualHubs = virtualHubs;
+    }
+
+    public void addNetworkExtensions(Network network) {
+        virtualHubs.forEach(vh -> addNetworkExtension(network, vh));
+    }
+
+    private void addNetworkExtension(Network network, VirtualHub virtualHub) {
+        Optional<Bus> bus = findBusById(network, virtualHub.getNodeName());
```

Comment:
Perhaps creating a map could be more performant than iterating for each virtual hub ?

Reply:
I can do that, but maybe one precision first : it would be a map of all the buses in the network then. And the number of buses in the network is quite big compared to number of virtual hubs (in CORE with have around 30 virtual hubs).

Is it relevant despite this ? Wouldn't we spend too much time building this big map while we juste need a few entry ? (I have no idea ^^)